File: sorting/selection-sort/selectionSort.py
# SelectionSort | O(N^2)
def selection_sort(list):
    for i in range(len(list) - 1):
        # Getting the lowest index (so far).
        lowest_index = i
        for j in range(i+1, len(list)):
            if list[j] < list[lowest_index]:
                # As the value in index "j" is lower than in the index "lowestIndex",
                # the "lowestIndex" is now "j".
                lowest_index = j
        if lowest_index != i:
            # As the "lowestIndex" is now different than "i",
            # we swap the values.
            list[lowest_index], list[i] = list[i], list[lowest_index]
    return list


# Testing the algorithm...
arr = [7, 5, 3, 9, 1, 6, 4, 2, 8]
print(f'Original: {arr}')
print(f'Sorted..: {selection_sort(arr)}')

File: sorting/selection-sort/selectionSort.js
// SelectionSort | O(N^2)
function selectionSort(array) {
     // Setting a temporary variable to swap values.
    let temp;  
    for (let i=0; i<array.length - 1; i++) {
        // Getting the lowest index (so far).
        let lowestIndex = i;    
        for (let j=i+1; j<array.length; j++) {
            if (array[j] < array[lowestIndex]) {
                // As the value in index "j" is lower than in the index "lowestIndex",
                // the "lowestIndex" is now "j".
                lowestIndex = j;    
            }                       
        }
        if (lowestIndex != i) {  
            // As the "lowestIndex" is now different than "i",
            // we swap the values.
            temp = array[i];     
            array[i] = array[lowestIndex];
            array[lowestIndex] = temp;
        }
    }
    return array;
}

// Testing the algorithm...
let arr = [7, 5, 3, 9, 1, 6, 8, 4, 2];
console.log(`Original: [${arr}]`);
console.log(`Sorted..: [${selectionSort(arr)}]`);
